samples/AzureFunctionsUnitTests/SampleUnitTests.cs

```diff
@@ -0,0 +1,194 @@
+
+        // Create dummy request object
+        var request = new TestRequestData(context);
+
+        // Invoke the function
+        var output = await AzureFunctionsApp.HttpStart(request, client, context);
+
+        // Assert expected logs are emitted
+        var capturedLogs = logger.CapturedLogs;
+        Assert.Contains(capturedLogs, log => log.Contains($"Started orchestration with ID = '{instanceId}'"));
+
+        // deserialize http output
+        output.Body.Seek(0, SeekOrigin.Begin);
+        using var reader = new StreamReader(output.Body, Encoding.UTF8);
+        string content = reader.ReadToEnd();
+        Dictionary<string, string>? keyValuePairs = JsonSerializer.Deserialize<Dictionary<string, string>>(content);
+
+        // Validate format of response URLs
+        Assert.NotNull(keyValuePairs);
+        Assert.Contains(keyValuePairs, kvp => kvp.Key == "id" && kvp.Value == instanceId);
+        Assert.Contains(keyValuePairs, kvp => kvp.Key == "purgeHistoryDeleteUri" && kvp.Value == $"http://localhost:8888/runtime/webhooks/durabletask/instances/{instanceId}");
+        Assert.Contains(keyValuePairs, kvp => kvp.Key == "sendEventPostUri" && kvp.Value == $"http://localhost:8888/runtime/webhooks/durabletask/instances/{instanceId}/raiseEvent/{{eventName}}");
+        Assert.Contains(keyValuePairs, kvp => kvp.Key == "statusQueryGetUri" && kvp.Value == $"http://localhost:8888/runtime/webhooks/durabletask/instances/{instanceId}");
+
+    }
+
+    // naive implementation of HttpRequestData for testing purposes
+    public class TestRequestData : HttpRequestData
+    {
+
+        readonly FunctionContext context;
+
+        public TestRequestData(FunctionContext functionContext) : base(functionContext)
+        {
+            this.context = functionContext;
+        }
+
+        public override Stream Body => new MemoryStream();
+
+        public override HttpHeadersCollection Headers => new HttpHeadersCollection();
+
+        public override IReadOnlyCollection<IHttpCookie> Cookies => new List<IHttpCookie>();
+
+        public override Uri Url => new Uri("http://localhost:8888/myUrl");
+
+        public override IEnumerable<ClaimsIdentity> Identities => new List<ClaimsIdentity>();
+
+        public override string Method => "POST";
+
+        public override HttpResponseData CreateResponse()
+        {
+            return new TestResponse(this.context);
+        }
+    }
+
+    // naive implementation of HttpResponseData for testing purposes, creating by TestRequestData's `CreateResponse` method
+    public class TestResponse : HttpResponseData
+    {
+        HttpStatusCode code = HttpStatusCode.NotFound;
+        Stream body = new MemoryStream();
+        HttpHeadersCollection headers = new HttpHeadersCollection();
+
+        public TestResponse(FunctionContext functionContext) : base(functionContext)
+        {
+        }
+
+        public override HttpStatusCode StatusCode { get => this.code; set => this.code = value; }
```

Comment:
Do this need fields? Does auto-properties `get; set;` not work?

Reply:
My IDE was seemingly complaining when I first tried it, but works now, so pushed it!